engine/pkg/meta/internal/etcdkv/connection.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdkv
+
+import (
+	"sync"
+
+	"github.com/pingcap/log"
+	metaModel "github.com/pingcap/tiflow/engine/pkg/meta/model"
+	"github.com/pingcap/tiflow/pkg/errors"
+	cerrors "github.com/pingcap/tiflow/pkg/errors"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/atomic"
+)
+
+// NewClientConnEtcdImpl return a new clientConnEtcdImpl
+func NewClientConnEtcdImpl() *clientConnEtcdImpl {
+	return &clientConnEtcdImpl{
+		isInitialized: atomic.NewBool(false),
+		tp:            metaModel.UnknownKVClientType,
+	}
+}
+
+type clientConnEtcdImpl struct {
+	mu            sync.Mutex
+	isInitialized *atomic.Bool
+	cli           *clientv3.Client
+	tp            metaModel.ClientType
+}
+
+// Initialize implements Initialize of ClientConn
+func (cc *clientConnEtcdImpl) Initialize(storeConf *metaModel.StoreConfig) error {
+	if storeConf == nil {
+		return cerrors.ErrMetaParamsInvalid.GenWithStackByArgs("storeConf is nil")
+	}
+
+	if storeConf.StoreType != metaModel.StoreTypeEtcd {
+		return cerrors.ErrMetaParamsInvalid.GenWithStackByArgs("unmatch store type,"+
+			"etcd conn but get:%s", storeConf.StoreType)
```

Comment:
The error will be `[DFLOW:ErrMetaParamsInvalid]meta params invalid:unmatch store type,etcd conn but get:%s%!(EXTRA string=<storeConf.StoreType>)`